### PR TITLE
Update "four_grids" gallery example

### PR DESCRIPTION
**Pull request by rossbar**
* Prevent node clipping.
 * Add slightly more descriptive intro to the example.
 * Switch from `plt.subplot` mechanism to using the `ax=` kwarg of nx.draw to show how to populate existing axes instances

### CODE REVIEW

**Comment by jarrodmillman** [comment]
How about using the Tableau Palette?
https://matplotlib.org/3.3.1/gallery/color/named_colors.html

**Comment by rossbar** [comment]
[Original version](https://networkx.github.io/documentation/latest/auto_examples/drawing/plot_four_grids.html) | [Version in this PR](https://499-890377-gh.circle-artifacts.com/0/doc/build/html/auto_examples/drawing/plot_four_grids.html#sphx-glr-auto-examples-drawing-plot-four-grids-py)

File: examples/drawing/plot_four_grids.py [hunk 1]
```diff
@@ -3,27 +3,40 @@
...
+ax = all_axes.flat
+
+nx.draw(G, pos, ax=ax[0], font_size=8)
+nx.draw(G, pos, ax=ax[1], node_size=0, with_labels=False)
+nx.draw(
+    G,
+    pos,
+    ax=ax[2],
+    node_color="tab:green",
+    edge_color="tab:gray",
```
Comment (jarrodmillman) [inline examples/drawing/plot_four_grids.py:30]: Add
```
    edgecolors="tab:gray",
```

File: examples/drawing/plot_four_grids.py [hunk 2]
```diff
@@ -3,27 +3,40 @@
...
+    node_size=250,
+    with_labels=False,
+    width=6,
+)
 H = G.to_directed()
-nx.draw(H, pos, node_color="b", node_size=20, with_labels=False)
+nx.draw(H, pos, ax=ax[3], node_color="tab:orange", node_size=20, with_labels=False)
```
Comment (jarrodmillman) [inline examples/drawing/plot_four_grids.py:36]:
```
nx.draw(
    H,
    pos,
    ax=ax[3],
    node_color="tab:orange",
    node_size=20,
    with_labels=False,
    arrowsize=10,
    width=2,
)
```

**Review by jarrodmillman — CHANGES_REQUESTED** [review]

File: examples/drawing/plot_four_grids.py (hunk 1 above)
Comment (rossbar) [reply examples/drawing/plot_four_grids.py:30]: Is this different from `edge_color="tab:gray"`? If I use "edgecolors" as the kwarg name instead it reverts back to black edges.

Comment (rossbar) [reply examples/drawing/plot_four_grids.py:30]: Oh nvm - I see now, it's the node edgecolor

**Pull request merged**